tests/test_deserialization.py

```diff
@@ -565,10 +581,27 @@ def test_timestamp_field_deserialization(self, fmt, value, expected):
         ["", "!@#", 0, -1, dt.datetime(2013, 11, 10, 1, 23, 45)],
     )
     def test_invalid_timestamp_field_deserialization(self, fmt, in_value):
-        field = fields.DateTime(format="timestamp")
+        field = fields.DateTime(format=fmt)
         with pytest.raises(ValidationError, match="Not a valid datetime."):
             field.deserialize(in_value)
 
+    @pytest.mark.parametrize("fmt", ["timestamp", "timestamp_ms"])
+    def test_oversized_timestamp_field_deserialization(self, fmt):
+        field = fields.DateTime(format=fmt)
+        with pytest.raises(ValidationError, match="Not a valid datetime."):
+            field.deserialize(999999999999999999999999)
+
+    #  Regression test for https://github.com/marshmallow-code/marshmallow/pull/2102
+    @pytest.mark.parametrize("fmt", ["timestamp", "timestamp_ms"])
+    @pytest.mark.parametrize(
+        "mock_fromtimestamp", [MockDateTimeOSError, MockDateTimeOverflowError]
```

Comment:
Do we need `MockDateTimeOverflowError`? Isn't this case covered by `test_oversized_timestamp_field_deserialization` above?

Reply:
`test_oversized_timestamp_field_deserialization` will most probably raise OverflowError internally, but they is not guarantee, it could also be OSError depending on the system.  

But indeed it is redundant as it was written before the mock. I decided to delete the first test and keep the two mocks as it make each error case explicit.